lib/pluginmanager/pack.rb

```diff
@@ -0,0 +1,60 @@
+require "bootstrap/util/compress"
+require "fileutils"
+
+class LogStash::PluginManager::Pack < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "compress package as a tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "compress package as a zip file", :default => LogStash::Environment.windows?
+  option "--[no-]clean", :flag, "clean up the generated dump of plugins", :default => true
+  option "--force", :flag, "Overwrite a previously generated package file", :default => false
+
+  def execute
+    puts("Packaging plugins for offline usage")
+
+    validate_target_file
+    LogStash::Bundler.invoke!({:package => true, :all => true})
+    compress_cache(LogStash::Environment::CACHE_PATH)
+    FileUtils.rm_rf(LogStash::Environment::CACHE_PATH) if clean?
+
+    puts("Generated at #{target_file}")
+  end
+
+  private
+
+  def compress_cache(location)
+    if zip?
+      LogStash::Util::Zip.compress(location, target_file)
+    else
+      LogStash::Util::Tar.compress(location, target_file)
+    end
```

Comment:
Both the `unpack.rb` and the `pack.rb` are using the same code to detect which class they should use to work with the archive. If we create a base class we could have a method in the parent class that will return the right format.

```ruby
def archive_manager
  zip? ? LogStash::Util::Zip : LogStash::Util::Tar
end
```